numba/lowering.py

```diff
@@ -247,7 +257,13 @@ def lower_inst(self, inst):
             return impl(self.builder, (target, index, value))
 
         elif isinstance(inst, ir.Del):
-            pass
+            try:
+                # XXX: incorrect Del injection?
+                val = self.loadvar(inst.value)
```

Comment:
IIUC, loadvar is creating a new load instruction that is not used - if this is the case, would calling getvar be enough to determine whether the `KeyError` is thrown?